CONTRIBUTING.md

```diff
@@ -108,36 +108,91 @@ Items maked a `*` will need work on the frontend as well.
+
+
+### Install `Homebrew`
+
+Following the [Homebrew install guide here](https://brew.sh/).
+* Open your terminal, copy and paste this command into your Terminal:
+`/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`
+
+## Project setup
+
+First, clone this project from Github
+
+```
+git clone https://github.com/emhoracek/smooch.git
+cd smooch
+```
+![alt text](https://preview.ibb.co/exRHQb/gitclone.png "Git clone repo")
+
+
 ### Install `cel2pnm`
+Now change to the `cel2pnm` directory.
 
-You'll need a C compiler to compile the program.
+* Now that you are in the top-level folder of this project, change into the `cel2pnm` directory: `cd cel2pnm`
+* Compile Smooch's `cel2pnm` program by running this command:
 
-On Mac OSX you can use [Homebrew] to install `gcc` with `brew install gcc`
+```
+gcc cel2pnm.c -o cel2pnm
+```
+![alt text](https://preview.ibb.co/npWqdw/compile_smooch.png "Compile cel2pnm program")
 
-Now change to the `cel2pnm` directory. Run `gcc cel2pnm.c -o cel2pnm`. That will
-result in the compiled program, `cel2pnm`.
+* Put `cel2pnm` in your [$PATH](https://askubuntu.com/questions/551990/what-does-path-mean).
+
+### Install `stack`
+
+You'll need `stack` to build Smooch. [`stack`](https://github.com/commercialhaskell/stack) is a Haskell dependency management
+tool (kinda like `npm` for JavaScript). 
+
+* Following the `stack` [install guide here](https://docs.haskellstack.org/en/stable/README/#how-to-install)
+* Copy and paste this command into your Terminal:
+
+```
+curl -sSL https://get.haskellstack.org/ | sh
+```
 
-Put `cel2pnm` in your PATH.
+![alt text](https://preview.ibb.co/cYhc8w/install_stack.png "Install stack")
 
-(A really helpful contribution would be to make this section irrelevant by
-moving the functionality of `cel2pnm` to JavaScript or Haskell! Or, providing
-more documentation of how to compile and install `cel2pnm` on different
-platforms.)
+* Once you have `stack` installed, change to the `app` directory and run `stack setup`. 
+This will install the correct Haskell version (this will take a while if you don't already have it).
 
-### Build Smooch
+![alt text](https://preview.ibb.co/dNNoFb/stack_setup.png "stack setup")
 
-You'll need `stack` to build Smooch. `stack` is a Haskell dependency management
-tool (kinda like `npm` for JavaScript). You can learn how to install it for
-various platforms in the [`stack` documentation].
+* Next, run `stack build` which will install project dependencies
+![alt text](https://preview.ibb.co/iC64ow/stack_build.png "stack build")
 
-Once you have `stack` installed, change to the `haskell` directory and run
-`stack setup`. That will install the correct Haskell version (this will take a
-while if you don't already have it).
+* Next, run `stack install rivet-autoimporter`
 
-Next, run `stack build`. That will compile the Smooch app.
+That will compile the Smooch app.
```

Comment:
`stack build` is the part that compiles the Smooch app, `rivet-autoimporter` is just another dependency, a tool for migrating the database. So, I think the `stack installl rivet-autoimporter` step should go above the `stack setup` step so this sentence makes more sense.